src/webots/control/WbController.cpp

```diff
@@ -251,9 +251,10 @@ void WbController::start() {
         warn(tr("Environment variables from runtime.ini could not be loaded: the file contains illegal definitions."));
       else {
         for (int i = 0; i < iniParser.size(); ++i) {
-          if (iniParser.sectionAt(i).compare("environment variables with relative paths") != 0 &&
-              iniParser.sectionAt(i).compare("environment variables for Linux") != 0 &&
-              iniParser.sectionAt(i).compare("environment variables for Linux 64") != 0)
+          if ((iniParser.sectionAt(i).compare("environment variables with relative paths") ||
+               iniParser.sectionAt(i).compare("environment variables with paths")) &&
```

Comment:
`(iniParser.sectionAt(i).compare("environment variables with relative paths") || iniParser.sectionAt(i).compare("environment variables with paths")` is always true.